exec/cnex/lib/posix.c

```diff
@@ -0,0 +1,781 @@
+    char *name1 = string_asCString(top(exec->stack)->string); pop(exec->stack);
+    Number fd1 = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(linkat(number_to_sint32(fd1), name1, number_to_sint32(fd2), name2, number_to_sint32(flag))));
+    free(name1);
+    free(name2);
+}
+#endif
+
+void posix_lockf(struct tagTExecutor *exec)
+{
+    Number size = top(exec->stack)->number; pop(exec->stack);
+    Number function = top(exec->stack)->number; pop(exec->stack);
+    Number fildes = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(lockf(number_to_sint32(fildes), number_to_sint32(function), number_to_uint64(size))));
+}
+
+void posix_lseek(struct tagTExecutor *exec)
+{
+    Number whence = top(exec->stack)->number; pop(exec->stack);
+    Number offset = top(exec->stack)->number; pop(exec->stack);
+    Number fildes = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(lseek(number_to_sint32(fildes), number_to_sint64(offset), number_to_sint32(whence))));
+}
+
+void posix_mkdir(struct tagTExecutor *exec)
+{
+    Number mode = top(exec->stack)->number; pop(exec->stack);
+    char *path = string_asCString(top(exec->stack)->string); pop(exec->stack);
+
+    push(exec->stack, wrap(mkdir(path, number_to_sint32(mode))));
+    free(path);
+}
+
+#if HAVE_AT_FUNCTIONS
+void posix_mkdirat(TExecutor *exec)
+{
+    Number mode = top(exec->stack)->number; pop(exec->stack);
+    char *path = string_asCString(top(exec->stack)->string); pop(exec->stack);
+    Number fd = top(exec->stack)->number; pop(exec->stack);
+
+    push(exec->stack, wrap(mkdirat(number_to_sint32(fd), path, number_to_sint32(mode))));
+    free(path);
+}
+#endif
+
+void posix_mkfifo(struct tagTExecutor *exec)
+{
+    Number mode = top(exec->stack)->number; pop(exec->stack);
+    char *path = string_asCString(top(exec->stack)->string); pop(exec->stack);
+
+    push(exec->stack, wrap(mkfifo(path, number_to_sint32(mode))));
+    free(path);
+}
+
+void posix_open(struct tagTExecutor *exec)
+{
+    Number mode = top(exec->stack)->number; pop(exec->stack);
+    Number oflag = top(exec->stack)->number; pop(exec->stack);
+    char *path = string_asCString(top(exec->stack)->string); pop(exec->stack);
+
+    push(exec->stack, wrap(open(path, number_to_sint32(oflag), number_to_sint32(mode))));
+    free(path);
+}
+
+void posix_pathconf(struct tagTExecutor *exec)
+{
+    Number name = top(exec->stack)->number; pop(exec->stack);
+    char *path = string_asCString(top(exec->stack)->string); pop(exec->stack);
+
+    push(exec->stack, wrap(pathconf(path, number_to_sint32(name))));
+    free(path);
+}
+
+void posix_pipe(struct tagTExecutor *exec)
+{
+    int fds[2];
+    int r = pipe(fds);
+    if (r < 0) {
+        push(exec->stack, wrap(r));
+        return;
+    }
+    Cell *rfd = cell_fromNumber(number_from_sint32(fds[0]));
+    Cell *wfd = cell_fromNumber(number_from_sint32(fds[1]));
+
+    push(exec->stack, wrap(r));
```

Comment:
This call to `wrap()` is not necessary here, because you've already checked the return value above.

Reply:
Fixed.